pkg/prom/instance/instance.go

```diff
@@ -552,6 +630,13 @@ func (i *Instance) getRemoteWriteTimestamp() int64 {
 	return ts * 1000
 }
 
+// Config returns the current Config stored by the Instance.
+func (i *Instance) Config() Config {
+	i.cfgMutex.Lock()
+	defer i.cfgMutex.Unlock()
+	return i.cfg
```

Comment:
Just a naive question. Why is the lock required here? Isn't `cfg` a reference?

Reply:
`i.cfg` will be copied by value here, but the mutex is needed to prevent `i.cfg` being read in the middle of a write from another Goroutine (i.e., the one calling `Update`), which is undefined behavior in Go.

Reply:
Ah, got it. Thanks :slightly_smiling_face: